src/nvim/ex_cmds.c

```diff
@@ -5846,3 +5910,298 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
+                        klist_t(matchedline_T) *lmatch, // NOLINT
+                        bool split)
+FUNC_ATTR_NONNULL_ARG(1, 2, 3)
+{
+  garray_T winsizes;
+  int save_restart_edit = restart_edit;
+  int save_State = State;
+  int save_exmode = exmode_active;
+  int save_cmdmsg_rl = cmdmsg_rl;
+
+  // Can't do this recursively.  Can't do it when typing a password.
+  if (cmdline_star > 0) {
+    beep_flush();
+    return;
+  }
+
+  // Save current window sizes.
+  win_size_save(&winsizes);
+
+  // Save the current window to restore it later
+  win_T *oldwin = curwin;
+
+  if (split) {
+    // Don't execute autocommands while creating the window.
+    block_autocmds();
+    // don't use a new tab page
+    cmdmod.tab = 0;
+
+    // Create a window for the command-line buffer.
+    if (win_split((int)p_cwh, WSP_BOT) == FAIL) {
+      beep_flush();
+      unblock_autocmds();
+      return;
+    }
+    cmdwin_type = get_cmdline_type();
+
+    // Create the command-line buffer empty.
+    (void)do_ecmd(0, NULL, NULL, NULL, ECMD_ONE, ECMD_HIDE, NULL);
+    (void)setfname(curbuf, (char_u *) "[live_sub]", NULL, true);
+    set_option_value((char_u *) "bt", 0L, (char_u *) "nofile", OPT_LOCAL);
+    set_option_value((char_u *) "swf", 0L, NULL, OPT_LOCAL);
+    curbuf->b_p_ma = false;  // Not Modifiable
+    curwin->w_p_fen = false;
+    curwin->w_p_rl = cmdmsg_rl;
+    cmdmsg_rl = false;
+    RESET_BINDING(curwin);
+
+    // Do execute autocommands for setting the filetype (load syntax).
+    unblock_autocmds();
+
+    // Showing the prompt may have set need_wait_return, reset it.
+    need_wait_return = false;
+
+    // Reset 'textwidth' after setting 'filetype'
+    // (the Vim filetype plugin sets 'textwidth' to 78).
+    curbuf->b_p_tw = 0;
+
+    // Save the buffer used in the split
+    livebuf = curbuf;
+
+    // Initialize line and highlight variables
+    int line = 0;
+    int src_id_highlight = 0;
+    long sub_size = STRLEN(sub);
+    long pat_size = STRLEN(pat);
+
+    // Get the width of the column which display the number of the line
+    linenr_T highest_num_line = 0;
+    kl_iter(matchedline_T, lmatch, current) {
+      highest_num_line = (*current)->data.lnum;
+    }
+
+    // computing the length of the column that will display line number
+    //TODO: Is log10 efficient for counting the nr of digits?
+    int col_width = log10(highest_num_line) + 1 + 3;
+    // better? log10 might not be precise for large numbers (ref?)
+    //int col_width = snprintf(NULL, 0, "%ld", highest_num_line) + 3;
+
+    // allocate a line sized for the window
+    char *str = xcalloc((size_t)curwin->w_frame->fr_width, sizeof(char));
+
+    // Append the lines to our buffer
+    kl_iter(matchedline_T, lmatch, current) {
+      matchedline_T mat = (*current)->data;
+      size_t line_size = STRLEN(mat.line) + col_width + 1;
+
+      // Reallocation if str not long enough
+      if (line_size > curwin->w_frame->fr_width * sizeof(char)) {
```

Comment:
Thank you! I threw in a `sizeof(char)` into the `xrealloc` for good measure :)

Do you have any stance on using `snprintf(NULL, 0, "%ld", highest_num_line) + 3;` instead of `log10(highest_num_line) + 1 + 3;` as I indicated in the comments?